### PR TITLE
e2e: override-frontend-port, duplicate path and multiple url path maps

### PR DESCRIPTION
**Pull request by akshaysngupta**
## Checklist
- [x] The title of the PR is clear and informative
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] Issues addressed by the PR are mentioned in the description followed by `Fixes`.

## Description

This PR adds e2e for two cases:
1. Test out the override-frontend port annotation and make a request at that port
2. Test the fixes for configuration reliability
    a. when ingress contains duplicate paths.
    b. when ingress contains multiple rules with multiple paths. This was failing before due to conflicting naming.
    c. when two pods in a service different container ports causing conflicts.

### CODE REVIEW

File: scripts/e2e/cmd/runner/testdata/one-namespace-one-ingress/override-frontend-port/app.yaml [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+---
+apiVersion: extensions/v1beta1
+kind: Ingress
+metadata:
+  name: ingress-https
+  annotations:
+    kubernetes.io/ingress.class: azure/application-gateway
+    appgw.ingress.kubernetes.io/backend-path-prefix: "/"
+    appgw.ingress.kubernetes.io/override-frontend-port: "8443"
```
Comment (3quanfeng) [inline scripts/e2e/cmd/runner/testdata/one-namespace-one-ingress/override-frontend-port/app.yaml:74]: can we test some edge case here like 65535 and 65536 or some reserved port such as 22?

File: scripts/e2e/cmd/runner/mfu_one_namespace_one_ingress_test.go [hunk 2]
```diff
@@ -189,6 +189,76 @@ var _ = Describe("MFU", func() {
...
+			publicIP, _ := getPublicIP(clientset, namespaceName)
+			Expect(publicIP).ToNot(Equal(""))
+
+			urlHttp := fmt.Sprintf("http://%s:%d/good", publicIP, 8080)
+			urlHttps := fmt.Sprintf("https://%s:%d/good", publicIP, 8443)
+			// http get to return 200 ok
+			_, err = makeGetRequest(urlHttp, "app.http", 200, true)
+			Expect(err).To(BeNil())
+			// https get to return 200 ok
+			_, err = makeGetRequest(urlHttps, "app.https", 200, true)
```
Comment (3quanfeng) [inline scripts/e2e/cmd/runner/mfu_one_namespace_one_ingress_test.go:220]: don't see the host "app.https" defined anywhere

File: scripts/e2e/cmd/runner/mfu_one_namespace_one_ingress_test.go [hunk 3]
```diff
@@ -189,6 +189,76 @@ var _ = Describe("MFU", func() {
...
+			Expect(err).To(BeNil())
+			Expect(readBody(resp)).To(Equal("app"))
+
+			url = fmt.Sprintf("http://%s/app", publicIP)
+			resp, err = makeGetRequest(url, "app.http", 200, true)
+			Expect(err).To(BeNil())
+			Expect(readBody(resp)).To(Equal("app"))
+
+			url = fmt.Sprintf("http://%s/app1", publicIP)
+			resp, err = makeGetRequest(url, "app.http", 200, true)
```
Comment (3quanfeng) [inline scripts/e2e/cmd/runner/mfu_one_namespace_one_ingress_test.go:257]: is it basing on first come first serve when same port name defined for two different ports? how does user know they have such invalid config and go fix by themself?

File: scripts/e2e/cmd/runner/testdata/one-namespace-one-ingress/override-frontend-port/app.yaml (hunk 1 above)
Comment (akshaysngupta) [reply scripts/e2e/cmd/runner/testdata/one-namespace-one-ingress/override-frontend-port/app.yaml:74]: sure!

File: scripts/e2e/cmd/runner/mfu_one_namespace_one_ingress_test.go (hunk 3 above)
Comment (akshaysngupta) [reply scripts/e2e/cmd/runner/mfu_one_namespace_one_ingress_test.go:257]: yes

File: scripts/e2e/cmd/runner/testdata/one-namespace-one-ingress/override-frontend-port/app.yaml (hunk 1 above)
Comment (akshaysngupta) [reply scripts/e2e/cmd/runner/testdata/one-namespace-one-ingress/override-frontend-port/app.yaml:74]: This test covers that scenario.
https://github.com/Azure/application-gateway-kubernetes-ingress/blob/9d9c12709e2bde5c7cc667ec7353dcbb1a4cb6b6/pkg/appgw/requestroutingrules_test.go#L308

Comment (akshaysngupta) [reply scripts/e2e/cmd/runner/testdata/one-namespace-one-ingress/override-frontend-port/app.yaml:74]: also, we don't have validations on using a restricted port from appgw side. If such ports are used, NRP will throw a validation error.

File: scripts/e2e/cmd/runner/mfu_one_namespace_one_ingress_test.go (hunk 3 above)
Comment (akshaysngupta) [reply scripts/e2e/cmd/runner/mfu_one_namespace_one_ingress_test.go:257]: we emit error messages when we find a conflict.

File: scripts/e2e/cmd/runner/mfu_one_namespace_one_ingress_test.go [hunk 4]
```diff
@@ -189,6 +189,76 @@ var _ = Describe("MFU", func() {
...
+			urlHttps := fmt.Sprintf("https://%s:%d/good", publicIP, 8443)
+			// http get to return 200 ok
+			_, err = makeGetRequest(urlHttp, "app.http", 200, true)
+			Expect(err).To(BeNil())
+			// https get to return 200 ok
+			_, err = makeGetRequest(urlHttps, "app.https", 200, true)
+			Expect(err).To(BeNil())
+		})
+
+		It("[invalid-config] should be able to work with duplicate paths and multiple backend port", func() {
```
Comment (3quanfeng) [inline scripts/e2e/cmd/runner/mfu_one_namespace_one_ingress_test.go:224]: could we change it to a different name :), invalid-config sounds like we need to error out and fail the test case, might be helpful to understand by adding some more description to indicate why it return 200 in case of duplicate port name used

Comment (akshaysngupta) [reply scripts/e2e/cmd/runner/mfu_one_namespace_one_ingress_test.go:224]: sure

**Review by 3quanfeng — APPROVED** [review]
LGTM!
